jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -124,6 +124,7 @@ def _parse_image(app, file_obj, data_label, ext=None):
     data_iter = get_image_data_iterator(app, file_obj, data_label, ext=ext)
 
     for data, data_label in data_iter:
+        data.coords.bounding_box = None
```

Comment:
This is too high up. What if coords is FITS WCS or `None`? Should be done inside the ASDF parser.

Reply:
does it do harm to also ensure that extrapolation will always be allowed?  Or is the bounding-box used elsewhere?

Reply:
The CEERS example used to reproduce this isn't an ASDF file nor is `data.coords` GWCS...

Reply:
Do you use it for the "zoom correctly" algorithm? If not, then maybe this is okay. I think the reason they disallowed extrapolation is that if the transformation isn't linear, you will get inaccurate answer outside the bounding box. Since we already say we don't support distortion in a lot of places, probably good enough.

Reply:
OK, even if you want to attach this to FITS WCS, you need to check for `coords = None` case.

Reply:
> Do you use it for the "zoom correctly" algorithm?

Good point, that [uses pixels_bounds](https://github.com/spacetelescope/jdaviz/pull/1897/files#diff-344d1a6a603d06bc55d66dbcdbccf2b7462a42f96b10e275873a1470a1fed4c3R67), but seems to be affected by a change to `bounding_box`.  I think I can rewrite that to rely on the shape of the underlying data entry so it won't conflict with this.

Reply:
https://github.com/kecnry/jdaviz/commit/e454042e8a698aaba5a62da7d684f3dd0e9fd077 changes the zoom logic in #1908 to rely on the underlying data shape instead of the `pixels_bounds`, which I think should be even more general and avoids the need for one of the previous fallback cases.  I tested both PRs together and they seem to work as expected, but stress testing is welcomed!

Reply:
> I think the reason they disallowed extrapolation is that if the transformation isn't linear, you will get inaccurate answer outside the bounding box.

I think for image displaying, slight distortion is preferred to just cropping data.  For the mouseover... we could handle this within the mouseover plugin itself (start displaying less precision or a warning message when the coordinates are outside the range of the reference data).

Reply:
> For the mouseover... we could handle this within the mouseover plugin itself (start displaying less precision or a warning message when the coordinates are outside the range of the reference data).

If we wanted to do this (either in this PR or later), we could probably store the original bounding box somewhere (`data.coords._orig_bounding_box`, for example) and then read from that during the mouseover to display additional text and/or lowering the number of digits displayed.  Is there anywhere else in the app that would be affected (besides visualization)?  I guess technically anything that uses a subset based on the overlapping of multiple data layers - but I don't think that is a large of a concern since wysiwyg 🤷

Reply:
I like that idea, could color the mouseover output yellow or something to warn that it's outside the defined region.

Reply:
We currently don't display mouse values outside the data dimension anyway, right? I'd prefer we don't show anything at all instead of having to explain what yellow text means.

Reply:
This PR enables showing mouseover information for a layer that extends beyond the original bounding box of the reference data WCS (#1899).

Reply:
Oh, right... 😒

Reply:
In that case, if different color makes sense to you and has no performance penalty, I guess I can go with that idea. If that proves to be a challenge, then we'll have to settle for just a warning in doc.

Reply:
(see update in https://github.com/spacetelescope/jdaviz/pull/1908#issuecomment-1349704313)